module/documents/actor/trait.mjs

```diff
@@ -232,24 +369,73 @@ export function keyLabel(trait, key) {
+ * @param {boolean} [options.final=false]  If this choice is part of a list of other grants or choices,
+ *                                         is it in the final position?
  * @returns {string}
  */
-export function choiceLabel(trait, choice) {
-  // Select from any trait values
-  if ( !choice.pool ) {
-    return game.i18n.format("DND5E.TraitConfigChooseAny", {
-      count: choice.count,
-      type: traitLabel(trait, choice.count).toLowerCase()
+export function choiceLabel(choice, { only=false, final=false }={}) {
+  if ( !choice.pool.size ) return "";
+
+  // Single entry in pool
+  // { count: 3, pool: ["skills:*"] } -> any three skills
+  // { count: 3, pool: ["skills:*"] } (final) -> three other skills
+  if ( choice.pool.size === 1 ) {
+    return keyLabel(choice.pool.first(), {
+      count: (choice.count > 1 || !only) ? choice.count : null, final: final && !only
     });
   }
 
-  // Select from a list of options
-  const choices = choice.pool.map(key => keyLabel(trait, key));
   const listFormatter = new Intl.ListFormat(game.i18n.lang, { type: "disjunction" });
+
+  // Singular count
+  // { count: 1, pool: ["skills:*"] } -> any skill
+  // { count: 1, pool: ["thief", "skills:*"] } -> Thieves Tools or any skill
+  // { count: 1, pool: ["thief", "tools:artisan:*"] } -> Thieves' Tools or any artisan tool
+  if ( (choice.count === 1) && only ) {
+    return listFormatter.format(choice.pool.map(key => keyLabel(key)));
+  }
+
+  // Select from a list of options
+  // { count: 2, pool: ["thief", "skills:*"] } -> Choose two from thieves tools or any skill
+  const choices = choice.pool.map(key => keyLabel(key));
   return game.i18n.format("DND5E.TraitConfigChooseList", {
     count: choice.count,
     list: listFormatter.format(choices)
   });
 }
+
+/* -------------------------------------------- */
+
+/**
+ * Create a human readable description of trait grants & choices.
+ * @param {Set<string>} grants                       Guaranteed trait grants.
+ * @param {TraitChoice[]} [choices=[]]               Trait choices.
+ * @param {object} [options={}]
+ * @param {string} [options.choiceMode="inclusive"]  Choice mode.
+ * @returns {string}
+ */
+export function localizedList(grants, choices=[], { choiceMode="inclusive" }={}) {
+  const choiceSections = [];
+
+  for ( const [index, choice] of choices.entries() ) {
+    const final = choiceMode === "exclusive" ? false : index === choices.length - 1;
+    choiceSections.push(choiceLabel(choice, { final, only: !grants.size && choices.length === 1 }));
+  }
+
+  let sections = Array.from(grants).map(g => keyLabel(g));
+  if ( choiceMode === "inclusive" ) {
+    sections = sections.concat(choiceSections);
+  } else {
+    const choiceListFormatter = new Intl.ListFormat(game.i18n.lang, { style: "long", type: "disjunction" });
+    sections.push(choiceListFormatter.format(choiceSections));
+  }
+
+  const listFormatter = new Intl.ListFormat(game.i18n.lang, { style: "long", type: "conjunction" });
```

Comment:
We can use `game.i18n.getListFormatter` which will lazily cache these formatters though it's not really a big deal.

Reply:
Hey, new method I didn't know existed, great!